### PR TITLE
Todolist add and  remove

### PR DESCRIPTION
**Pull request by souad988**
Add add and remove functionalities 
Add the update task function

### CODE REVIEW

File: src/modules/task.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export default class Task {
+  constructor(description = '', index = 0) {
+    this.description = description;
+    this.completed = false;
+    this.index = index;
```
Comment (rloterh) [inline src/modules/task.js:5]: - [ ] New entries **do not match** the property index. The _property index_ must be set to the value of the _new array length_ (i.e. if a 5th task is added to the list, the index of that task should equal 5).

**Index mismatch in your application:**
image

File: src/index.js [hunk 2]
```diff
@@ -1,62 +1,92 @@
...
+        console.log(e.path[1].querySelector('.checkbox').id);
+        todolist.remove(e.path[1].querySelector('.checkbox').id);
+        populateStorage(todolist.tasks);
+        displayToDoTasks(tasksContainer);
+      });
+    }
+  });
+  const descriptionInputs = document.querySelectorAll('.description');
+  descriptionInputs.forEach((description) => description.addEventListener(('change'), (e) => {
+    todolist.update(e.target.value, e.path[1].querySelector('.checkbox').id);
```
Comment (rloterh) [inline src/index.js:68]: - [ ] Checkbox values are not being updated. Checkbox values remain _false_ in your local storage whether it's checked or unchecked. Kindly ensure that **all** changes to the To-Do List are saved in local storage.

File: dist/index.html [hunk 3]
```diff
@@ -1 +1,23 @@
...
+    <div class="todolist_container">
+      <div class="todolist_header"><h5>My Todolist</h5>
+        <i class="fas fa-sync"></i>
+     </div>  
+      <form class="add_task">
+          <input type="text" placeholder=" Add to list..." class="add_task_input"/>
+          <button class="add_task_button"><i class="fas fa-plus"></i></button>
+      </form>
+      <ul class="tasks"></ul>
+      <div class="clear">Clear all completed</div>
```
Comment (rloterh) [inline dist/index.html:20]: - [ ] OPTIONAL
The Clear all completed button is not functioning correctly. Nothing happens when this button is clicked. Kindly endeavor to improve upon this.

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Souad, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦  🚦  

## Good Points 👍  
- Descriptive Pull Request
- CRUD function working as intended
- No linter errors


## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

_Kindly consider my comments marked **[OPTIONAL]** under this review._

<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: dist/index.html (hunk 3 above)
Comment (souad988) [reply dist/index.html:20]: hello,
thanks for reviewing , in the requirements for day 3 all we asked to implement are add and remove functionalities

File: src/index.js (hunk 2 above)
Comment (souad988) [reply src/index.js:68]: thanks for reviewing 
those are the requirements for day 3 :
Remove all hardcoded items from the tasks array.
Create a new JavaScript file for the new functionality.
Implement a function for adding a new task (add a new element to the array).
Implement a function for deleting a task (remove an element from the array).
Implement a function for editing task descriptions.
By default new tasks should have the property completed set to false and the property index set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal to 5).
Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique.
All changes to the To Do List should be saved in local storage.

File: dist/index.html (hunk 3 above)
Comment (Whoistolu) [reply dist/index.html:20]: I see this is optional 👍🏽

File: src/modules/task.js (hunk 1 above)
Comment (Whoistolu) [reply src/modules/task.js:5]: I see that this has been fixed also 👍🏽

Comment (Whoistolu) [reply src/modules/task.js:5]: 👍🏽  I can see that this has been fixed also

File: src/index.js (hunk 2 above)
Comment (Whoistolu) [reply src/index.js:68]: I see that this is not a requirement for this particular milestone so we are good for now. 👍🏽

**Review by Whoistolu — APPROVED** [review]
### Hi @souad988,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**